### PR TITLE
[labs/analyzer] Factor Module, Classes, Variables into separate modules

### PR DESCRIPTION
**Pull request by kevinpschaaf**

### CODE REVIEW

File: packages/labs/analyzer/src/lib/javascript/variables.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @fileoverview
+ *
+ * Utilities for working with classes
```
Comment (sorvell) [inline packages/labs/analyzer/src/lib/javascript/variables.ts:10]:
```suggestion
 * Utilities for working with variables
```

File: packages/labs/analyzer/src/lib/javascript/classes.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (sorvell) [inline packages/labs/analyzer/src/lib/javascript/classes.ts:1]: Naming the containing folder "javascript" seems a little obtuse. Can you describe the contents more?

File: packages/labs/analyzer/src/lib/javascript/classes.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+/**
+ * @fileoverview
+ *
+ * Utilities for working with classes
+ */
+
+import ts from 'typescript';
+import {ClassDeclaration} from '../model.js';
```
Comment (sorvell) [inline packages/labs/analyzer/src/lib/javascript/classes.ts:14]: It seems a little strange that you're factoring out a factory for getting a `ClassDeclaration`, but the `ClassDeclaration` class itself is still defined in the monolithic "model.js" module. Should that be broken up as well and perhaps go here?

File: packages/labs/analyzer/src/lib/javascript/variables.ts (hunk 1 above)
Comment (sorvell) [inline packages/labs/analyzer/src/lib/javascript/variables.ts:10]:
```suggestion
 * Utilities for working with variables
```

**Review by sorvell — APPROVED** [review]
Seems minor but, just a few comments, consider optional. I'm also not super familiar with the code but no red flags.

File: packages/labs/analyzer/src/lib/analyzer.ts [hunk 4]
```diff
@@ -42,14 +30,15 @@ export class Analyzer {
...
     // to analyzePackage() or move it to an async factory function that
     // passes these to the constructor as arguments.
+    let packageJson;
     try {
-      this.packageJson = JSON.parse(
+      packageJson = JSON.parse(
         fs.readFileSync(path.join(packageRoot, 'package.json'), 'utf8')
       );
     } catch (e) {
       throw new Error(`package.json not found in ${packageRoot}`);
```
Comment (justinfagnani) [inline packages/labs/analyzer/src/lib/analyzer.ts:39]: btw, there's two potential errors here: no file and malformed JSON.

File: packages/labs/analyzer/src/lib/javascript/classes.ts (hunk 2 above)
Comment (justinfagnani) [reply packages/labs/analyzer/src/lib/javascript/classes.ts:1]: I might name it "typescript" - but the context is analysis utilities for TS, as opposed to possibly future utilities for custom-elements-manifest, CSS, or HTML.

File: packages/labs/analyzer/src/lib/javascript/classes.ts (hunk 3 above)
Comment (justinfagnani) [reply packages/labs/analyzer/src/lib/javascript/classes.ts:14]: I like a separation between the model and the analysis functions that produce them. We'll have other production code paths when we add custom-element-manifest into the mix soon.

**Review by justinfagnani — APPROVED** [review]
I'd name the folder "typescript" rather than "javascript". Otherwise, LGTM.

File: packages/labs/analyzer/src/lib/analyzer.ts (hunk 4 above)
Comment (kevinpschaaf) [reply packages/labs/analyzer/src/lib/analyzer.ts]: Fixed

**Pull request merged**